bundles/org.openhab.binding.tuya/src/main/java/org/openhab/binding/tuya/internal/handler/TuyaDeviceHandler.java

```diff
@@ -508,6 +546,25 @@ private void addChannels(Map<String, SchemaDp> schema) {
                 channeltypeUID = CHANNEL_TYPE_UID_NUMBER;
                 configuration.put("min", schemaDp.min);
                 configuration.put("max", schemaDp.max);
+
+                if (schemaDp.scale > 0 || !schemaDp.unit.isEmpty()) {
+                    channeltypeUID = CHANNEL_TYPE_UID_QUANTITY;
+
+                    if (!schemaDp.unit.isEmpty()) {
+                        try {
+                            QuantityType<?> reftype = new QuantityType<>("1 " + schemaDp.unit);
```

Comment:
use UnitUtils.parseUnit to directly parse the unit here, rather than using a throwaway dummy QuantityType